src/imitation/scripts/parallel.py

```diff
@@ -31,14 +29,12 @@ def parallel(
     init_kwargs: Mapping[str, Any],
     local_dir: Optional[str],
     upload_dir: Optional[str],
-    repeat: int = 1,
-    eval_best_trial: bool = False,
-    eval_best_trial_resource_multiplier: int = 1,
-    eval_trial_seeds: int = 5,
-    experiment_checkpoint_path: str = "",
-    syncer=None,
-    resume: Union[str, bool] = False,
-) -> None:
+    repeat: int,
+    search_alg: Optional[str],
```

Comment:
I would prefer if you did not introduce any tuning-specific arguments to the `parallel` script. One way to solve this is to add a `tune_run_kwargs` parameter and use that to inject the optuna search algo etc.

Then it becomes the responsibility of the `tuning.py` script to set everything up correctly.

Reply:
I think `repeat` is a good argument to keep in the `parallel` script because it allows a sampled hyperparameter config to run multiple times, which is a good general feature to have in the `parallel` script. So if we want to keep it in the `parallel` script, we must keep `search_alg` in the `parallel` script because `repeat` requires a search algorithm to be specified first.

If we don't think `repeat` is beneficial to keep in the `parallel` script, I can introduce a `tune_run_kwargs` in `parallel` and move the repeat and search algorithm options to the `tuning` script only.

Reply:
I agree with the `repeat` argument. However, if we just repeat the same configuration a couple of times, why do we need a search algorithm? Maybe the Ray API needs one up just any search algorithm will do right?

Reply:
> if we just repeat the same configuration a couple of times, why do we need a search algorithm? 

The Ray API for repeating takes a search algorithm and wraps a repeater on top of it to give the modified search algorithm: `search.Repeater(optuna.OptunaSearch(), repeat=repeat)`.

The repeater allows running multiple trials with the same configuration sampled by the algorithm. Hence, it needs the search algorithm as an argument. I don't think we can have the repeat argument without the search algorithm.

Reply:
What I meant was: we could just set the search algo to any simple search algo as soon as `repeat > 0` then we do not have to expose this quirk of the Ray API to our users.

Reply:
That makes sense. I have redacted `search_alg` from the arguments of the parallel script, along with some other rarely used arguments to introduce `tune_run_kwargs`. Let me know if this looks good.